be/src/exec/pipeline/pipeline_driver.cpp

```diff
@@ -317,12 +317,20 @@ void PipelineDriver::finalize(RuntimeState* runtime_state, DriverState state) {
         if (_query_ctx->count_down_fragments()) {
             auto query_id = _query_ctx->query_id();
             DCHECK(!this->is_still_pending_finish());
-            if (_fragment_ctx->enable_resource_group()) {
-                if (_workgroup) {
+
+            auto frag_id = _fragment_ctx->fragment_instance_id();
+            int active_fragments = _query_ctx->num_active_fragments();
+            int active_drivers = _fragment_ctx->num_drivers();
+            int64_t wg_id = _workgroup != nullptr ? _workgroup->id() : 0;
+            if (ExecEnv::GetInstance()->query_context_mgr()->remove(query_id)) {
+                if (wg_id != 0) {
```

Comment:
`0` is `WorkGroup::DEFAULT_WG_ID`, representing the default workgroup, so `_workgroup->id()` may return `0`. Maybe we could just check `_workgroup != nullptr` here.

Reply:
done